thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashService.js

```diff
@@ -79,11 +87,11 @@ HashService.prototype = {
       }
       var defaultValue = paramNamesToDefaultValuesMap[paramName];
       var value = this.get(paramName);
-      if (value == undefined && defaultValue != undefined) { //if default value is present, use that
+      if (!value && defaultValue) { //if default value is present, use that
```

Comment:
You can condense these lines 89-95 into one succinct `default` operation:

```javascript
var value = this.get(paramName) || this.set(paramName, defaultValue);
```

or  even 88 -95

```javascript
var value = this.get(paramName) || this.set(paramName, paramNamesToDefaultValuesMap[paramName]);
```

if the var `defaultValue` is not used anywhere else. `Ember.set` returns the passed in value